src/Agent.Service/Windows/AgentListener.cs

```diff
@@ -0,0 +1,276 @@
+using System;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+namespace AgentService
+{
+    public enum ExecutionMode
+    {
+        Process,
+        Service
+    }
+
+    public class AgentListener
```

Comment:
This is not the listener. This terminology is confusing given the existing meaning of "listener".